### DISCUSSION TITLE
✅ ♿ 🚀 Feature Request: Make the Accordion widget WCAG 2.1 compliant (Accessibility) - [ED-10524]

### DISCUSSION THREAD

**Discussion by ampersandblack** [elementor/elementor#15524] [Feature Request]
My site is been audited for Accessibility compliance. 2 pages that uses the accordion widget are getting this issue.

An element with a role that hides child elements contains focusable child elements. 
 This role element marks child elements as presentational, which hides them from the accessibility tree, but some of these children are focusable, so they can be navigated to, but are not voiced in a screen reader. 

  **role=tab role=tab role=tab role=tab role=tab** https://aguadorivas.ampersandstudios.club/communications/ 

  **role=tab role=tab role=tab role=tab role=tab** https://aguadorivas.ampersandstudios.club/it-networking/

I also have to say that the accordion cannot be navigated using the keyboard, after the first tab jumps to the next section. 

**ADA** and **WCAG** compliance is a must!!!! You can get sued and people don´t even know!!!

The tool I used for auditing is **Sortsite** (used by the US Department of Justice)

**Comment by Trott** [comment]
Providing more information:

When using accordion or toggle widgets with focusable content, accessibility issues are introduced by the presence of keyboard-focusable content (such as `<a>` elements) in elements where `role="tab"` is set, as it is in these widgets. There is not an accessibility issue if the content is, for example, a paragraph of text. But if the widget contains a link or other keyboard-focusable content, then we run into issues as described in:

* https://help.siteimprove.com/support/solutions/articles/80001058756-accessibility-rule-role-with-implied-hidden-content-has-keyboard-focus-explained
* https://www.w3.org/TR/WCAG/#name-role-value
* https://alfa.siteimprove.com/rules/sia-r90

I would like to be able to use an accordion or toggle-type widget that does not have this accessibility limitation with keyboard-focusable content. If the widget can be configured to use `<summary>` and `<details>` tags instead of `<div role="tab">`, then I believe that would resolve the issue. Alternatively, if a different widget can be made available that uses `<summary>` or `<details>`, that would work as well.

I've looked for other widgets that might be suitable within the Elementor widget collection but haven't found any.

As a workaround, I can do any of the following not-so-great options:

* Remove all keyboard-focusable content from the content of the toggle/accordion.
* Remove the widget and just have all the information display on the page (or maybe find a way to insert `<summary>` and `<details>` tags myself into the page).
* Replace the widget with a link to a separate page.

It would be great if the toggle and/or accordion widgets could be made to accommodate keyboard-focusable content in an accessible way.

**Comment by Trott** [comment]
With the recent change to `role="button"` for toggle, this is still an issue (at least as far as Siteimprove is concerned) because the `<a>` element that contains the title text has `tabIndex="0"` hardcoded and it is inside the `role="button"` `<div>`. The button itself should be focusable and the contents of the button should not be focusable.

I'm not sure if it has other negative side effects, but one simple solution is to remove the `tabIndex="0"` hardcoding [here](https://github.com/elementor/elementor/blob/fd743c6ff046f3e34a2c22d83e09df0b62ea9fd7/includes/widgets/toggle.php#L621) and [here](https://github.com/elementor/elementor/blob/fd743c6ff046f3e34a2c22d83e09df0b62ea9fd7/includes/widgets/toggle.php#L708).

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with the new Nested Accordion in Elementor v3.17! 🥳 

The new Nested Accordion is now fully accessible. We recommend that you replace all your accordions and toggles with the new Nested Accordion.

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
